federation/cmd/federation-controller-manager/app/controllermanager.go

```diff
@@ -154,44 +154,67 @@ func StartControllers(s *options.CMServer, restClientCfg *restclient.Config) err
 	glog.Infof("Loading client config for cluster controller %q", "cluster-controller")
 	ccClientset := federationclientset.NewForConfigOrDie(restclient.AddUserAgent(restClientCfg, "cluster-controller"))
 	glog.Infof("Running cluster controller")
-	go clustercontroller.NewclusterController(ccClientset, s.ClusterMonitorPeriod.Duration).Run()
-	dns, err := dnsprovider.InitDnsProvider(s.DnsProvider, s.DnsConfigFile)
-	if err != nil {
-		glog.Fatalf("Cloud provider could not be initialized: %v", err)
-	}
 
 	discoveryClient := discovery.NewDiscoveryClientForConfigOrDie(restClientCfg)
 	serverResources, err := discoveryClient.ServerResources()
 	if err != nil {
 		glog.Fatalf("Could not find resources from API Server: %v", err)
 	}
 
-	glog.Infof("Loading client config for namespace controller %q", "namespace-controller")
-	nsClientset := federationclientset.NewForConfigOrDie(restclient.AddUserAgent(restClientCfg, "namespace-controller"))
-	namespaceController := namespacecontroller.NewNamespaceController(nsClientset, dynamic.NewDynamicClientPool(restclient.AddUserAgent(restClientCfg, "namespace-controller")))
-	glog.Infof("Running namespace controller")
-	namespaceController.Run(wait.NeverStop)
+	go clustercontroller.NewclusterController(ccClientset, s.ClusterMonitorPeriod.Duration).Run()
+
+	if controllerEnabled(s.Controllers, serverResources, servicecontroller.ControllerName, servicecontroller.RequiredResources, true) {
+		dns, err := dnsprovider.InitDnsProvider(s.DnsProvider, s.DnsConfigFile)
```

Comment:
There isn't an open issue because merging this PR would cause the problem.  I really think the parameter needs to be made optional in the PR that allows the services controller to be disabled.